_posts/2022-04-26-fivetran-custom-connector.md

```diff
@@ -0,0 +1,338 @@
+---
+layout: post
+title: FiveTran Paginated Custom Connector
+date: 2022-05-26
+tags: fivetran custom connector custom-connector pagination getmore state-management
+author: Joseph Miocevich
+image: img/fivetran_logo.png
+description:
+  Overview of a custom fivetran connector with pagination
+---
+
+![FiveTran Paginated Custom Connectors](/img/fivetran_logo.png)
+
+<br/>
+
+Understanding and debugging your FiveTran custom connector is not straightfoward, with many online tutorials excluding the key component of data API's... pagination. After reading this blog you will have a better understanding of the expected behaviour between FiveTran and your custom connector when pagination is thrown into the mix.
+
+## Overview
+
+A FiveTran custom connector is a cloud hosted function that can be set up on a cloud platform of your choice, which allows you to load data into FiveTran with a API FiveTran does not natively support. 
+
+Like standard connectors, custom connectors have a number of benefits:
+>
+- Incremental updates
+- Source data type inference
+- Automatic schema updates
+- Data de-deduplication
+- Destination ingestion optimisations
+- Logs and alerts to monitor events and troubleshoot issues
+
+Use FiveTran's Function connectors if:
+>
+- Fivetran doesn't have a connector for your source
+- You are using private APIs or custom applications
+- You are using a source or API that Fivetran is unlikely to support in the near future
+- You want to sync unsupported file formats that require pre-processing
+- You have sensitive data that needs filtering or anonymizing before entering the destination
+
+*Fivetran* . Fivetran Custom Connector. (n.d.). Retrieved May 26, 2022, from https://fivetran.com/docs/functions 
+
+<br>
+
+***
+<br>
+## General Principles
+
+**State Management**
+
+State is a JSON object that contains cursors from the previous successful FiveTran executions run.
+
+[https://fivetran.com/docs/functions/faq/use-state-object](https://fivetran.com/docs/functions/faq/use-state-object) 
+
+[https://fivetran.com/docs/functions/faq/use-secrets-object](https://fivetran.com/docs/functions/faq/use-secrets-object)
+
+
+**Potential Issues to Consider**
+
+- Lambda timeouts and payload limit
```

Comment:
Add link to the relevant AWS documentation?

Reply:
Good idea

Reply:
added